src/network/networkd-link.c

```diff
@@ -1143,6 +1144,16 @@ void link_check_ready(Link *link) {
         if (link->state != LINK_STATE_CONFIGURED)
                 link_enter_configured(link);
 
+        /* Reconfigure any running DHCP servers with information from the new
+         * link if necessary. */
+        HASHMAP_FOREACH(carrier, link->manager->links, i) {
+                if (link_dhcp4_server_enabled(link) && (link->flags & IFF_UP)) {
```

Comment:
Maybe, `link_dhcp4_server_enabled(carrier)` and `(carrier->flags & IFF_UP)`.

Reply:
\o/